languages/python/reference/exercise-concepts/markdown.md

```diff
@@ -0,0 +1,48 @@
+# Concepts for markdown
+
+[Example Implementation](https://github.com/exercism/python/blob/master/exercises/markdown/example.py)
+
+## General  
+
+
+- functions: This exercise requires a re-write of pre-esiting code that uses functions to parse passed-in text in _markdown_.  Tests expect a function named `parse` that can be called to transform the _markdown_ formatted text and return HTML formatted text.
+
+- function arguments:  The example solution uses multiple functions that take various function arguments to operate on different pieces of the passed in markdown string.
+
+- `re` module:  Both the original code to be refactored for this exercise and the example solution import and use the `re` module for Regular Expressions in python.
+
+- `import` key word/ importing:  Both the original code to be refactored for the exercise and the example solution use the `import` keyword to import the `re` module in support of Regular Expressions in python.
+
+- `str.format()` :  The example solution uses `str.split()` to break the passed in markdown string into a list of lines broken up by the `\n` character.
+
+- `re.match()`:  Method of the `re` class that returns a `match` object with any matched values from a specified Regualr Expression or pre-compliled Regular Expression.  The example uses `re.match()` in multiple places to search for text patterns that need re-formatting or subsitituting. 
+
+- regular expressions:  As a seperate language with a seperate interpreter that is often embeded inside another programming language.  Like many other programming lanagues in use, python supports the Regular Expression language and implements a Regular Expression interpreter in the core `re` module.  Both the original code to be refactored for this exercise and the example solution use the core `re` module to access Regualr Expression functionality.
```

Comment:
Spelling: multiple instances of "seperate" -> "separate"; "embeded" -> "embedded", "Regualr" -> "Regular".

This might be a place to introduce some more precise terminology. Regular expressions are specifically a Domain Specific Language (DSL), and Python supports a specific dialect of that DSL (irritatingly, a unique quasi-dialect of PCRE that no one else supports in quite the same way, that's why the `regex` module is much nicer for interoperability with other general programming languages). Also there isn't actually a separate interpreter (which implies a distinct process), rather it's just a standard function call that accepts a funny-looking string.

Reply:
Bleah.  Hope this is better phrasing:

>- regular expressions:  A Domain Specific Language (DSL) for text processing. Like many other programming languages in use, python supports a quasi-dialect of PCRE (_Perl compatible regular expressions_).  `Regular expressions` can be used via the core python `re` module, or the third-party `regex` module.  Both the original code to be refactored for this exercise and the example solutions use the core `re` module to access `regular expressions` functionality.